app/assets/javascripts/Components/Result/marks_panel.jsx

```diff
@@ -361,7 +361,11 @@ class RubricCriterionInput extends React.Component {
   };
 
   render() {
-    const levels = [0, 1, 2, 3, 4].map(this.renderRubricLevel);
+    const level_marks = [];
+    for (let i = 0; i < this.props.levels.length; i++) {
+      level_marks.push(this.props.levels[i].mark);
+    }
+    const levels = level_marks.map(this.renderRubricLevel);
```

Comment:
Try changing `this.renderRubricLevel` to accept a "level" object rather than just a single number. This way you should be able to just do `this.props.levels.map(this.renderRubricLevel)`.